packages/core/src/new-api/types/execution-state.ts

```diff
@@ -120,32 +120,63 @@ export interface DeploymentExecutionState
   > {
   storedArtifactPath: string; // As stored in the deployment directory.
   storedBuildInfoPath?: string; // As stored in the deployment directory. Optional as it's not always present
+  contractName: string;
   constructorArgs: ArgumentType[];
   libraries: Record<string, string>; // TODO: Do we need to store their future ids for the reconciliation process?
   value: bigint;
+  from: string | undefined;
   contractAddress?: string; // The result
+}
 
-  // from: string; -- Will we support this in the beta? I'll ignore it for now, but we should decide it asap
+export interface CallExecutionState
+  extends BaseExecutionState<FutureType.NAMED_CONTRACT_CALL> {
+  contractAddress: string;
+  functionName: string;
+  args: ArgumentType[];
+  value: bigint;
+  from: string | undefined;
 }
 
-type CallExecutionState = BaseExecutionState<FutureType.NAMED_CONTRACT_CALL>;
+export interface StaticCallExecutionState
+  extends BaseExecutionState<FutureType.NAMED_STATIC_CALL> {
+  contractAddress: string;
+  functionName: string;
+  args: ArgumentType[];
+  result?: ArgumentType;
```

Comment:
We may need to use a Buffer here and store the non-decoded version, as we may get errors or invalid ABI encoded values. 

I guess we also want to store if the execution was successful or not. In the other execution states we can check if they failed by looking at their txs, but not in this case.

Reply:
> I guess we also want to store if the execution was successful or not
 self-answer: that's part of the `BaseExecutionState`